test/server_test.go

```diff
@@ -32,6 +32,41 @@ import (
 
 type ctxKey string
 
+// TestServerReturningContextError verifies that if a context error is returned
+// by the service handler, the status will have the correct status code, not
+// Unknown.
+func (s) TestServerReturningContextError(t *testing.T) {
+	ss := &stubserver.StubServer{
+		EmptyCallF: func(ctx context.Context, in *testpb.Empty) (*testpb.Empty, error) {
+			return nil, context.DeadlineExceeded
+		},
+		FullDuplexCallF: func(stream testpb.TestService_FullDuplexCallServer) error {
+			return context.DeadlineExceeded
+		},
+	}
+	if err := ss.Start(nil); err != nil {
+		t.Fatalf("Error starting endpoint server: %v", err)
+	}
+	defer ss.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	_, err := ss.Client.EmptyCall(ctx, &testpb.Empty{})
+	if s, ok := status.FromError(err); !ok || s.Code() != codes.DeadlineExceeded {
+		t.Fatalf("ss.Client.EmptyCall(ctx, _) = _, %v; want _, <status with Code()=DeadlineExceeded>", err)
```

Comment:
Can we avoid underscores in these error message. They make them really hard to read and add no value. I know this is the way many existing tests report errors, but do we need to continue doing it.

Reply:
Are you suggesting deleting `_,`?

Reply:
Yes, thanks :)

Reply:
> add no value

I have to disagree on this a bit.  They show the parameter/return signature of the function being called accurately.  If a function accepts 3 values and returns 3, but you only show 1 or 2 of each of them, how do you know what the values represent?

If you want to remove them here then don't imply the function only returns one value.  Do something like this instead:

```
t.Fatalf("ss.Client.EmptyCall() got error %v; want status with Code()=DeadlineExceeded", err)
```

Reply:
Done